apps/crn-server/src/handlers/calendar/subscribe-handler.ts

```diff
@@ -1,57 +1,60 @@
 import {
   AlertsSentry,
-  calendarCreatedHandlerFactory,
+  calendarCreatedHandlerFactory as calendarCreatedSquidexHandlerFactory,
+  calendarCreatedContentfulHandlerFactory,
   getJWTCredentialsFactory,
   subscribeToEventChangesFactory,
   unsubscribeFromEventChangesFactory,
 } from '@asap-hub/server-common';
-import { RestCalendar, SquidexGraphql, SquidexRest } from '@asap-hub/squidex';
 import * as Sentry from '@sentry/serverless';
 import 'source-map-support/register';
 import {
-  appName,
   asapApiUrl,
-  baseUrl,
+  contentfulEnvId,
+  contentfulSpaceId,
+  contentfulAccessToken,
   googleApiCredentialsSecretId,
   googleApiToken,
   googleApiUrl,
+  isContentfulEnabledV2,
   region,
 } from '../../config';
-import { CalendarSquidexDataProvider } from '../../data-providers/calendars.data-provider';
-import { getAuthToken } from '../../utils/auth';
+import { getCalendarDataProvider } from '../../dependencies/calendars.dependencies';
 import logger from '../../utils/logger';
 import { sentryWrapper } from '../../utils/sentry-wrapper';
 
-const squidexGraphqlClient = new SquidexGraphql(getAuthToken, {
-  appName,
-  baseUrl,
-});
-const calendarRestClient = new SquidexRest<RestCalendar>(
-  getAuthToken,
-  'calendars',
-  { appName, baseUrl },
-);
-const calendarDataProvider = new CalendarSquidexDataProvider(
-  calendarRestClient,
-  squidexGraphqlClient,
-);
 /* istanbul ignore next */
 const getJWTCredentialsAWS = getJWTCredentialsFactory({
   googleApiCredentialsSecretId,
   region,
 });
-const webhookHandler = calendarCreatedHandlerFactory(
+
+export const calendarCreatedHandlerFactory = isContentfulEnabledV2
+  ? calendarCreatedContentfulHandlerFactory
+  : calendarCreatedSquidexHandlerFactory;
+
+export const contentfulDeliveryApiConfig = {
+  environment: contentfulEnvId,
+  space: contentfulSpaceId,
+  accessToken: contentfulAccessToken,
+};
+
+export const path = isContentfulEnabledV2 ? 'contentful' : undefined;
+
+export const webhookHandler = calendarCreatedHandlerFactory(
   subscribeToEventChangesFactory(getJWTCredentialsAWS, logger, {
     asapApiUrl,
     googleApiToken,
     googleApiUrl,
+    path,
   }),
   unsubscribeFromEventChangesFactory(getJWTCredentialsAWS, logger, {
     googleApiUrl,
   }),
-  calendarDataProvider,
+  getCalendarDataProvider(),
   new AlertsSentry(Sentry.captureException.bind(Sentry)),
   logger,
+  contentfulDeliveryApiConfig,
 );
```

Comment:
About this one, looking at the last version of this PR, the `apps/crn-server/src/handlers/calendar/subscribe-handler.ts` would change to

```
export const webhookHandler = isContentfulEnabledV2
  ? calendarCreatedContentfulHandlerFactory(
      subscribeToEventChangesFactory(getJWTCredentialsAWS, logger, {
        asapApiUrl,
        googleApiToken,
        googleApiUrl,
        cms: 'contentful',
      }),
      unsubscribeFromEventChangesFactory(getJWTCredentialsAWS, logger, {
        googleApiUrl,
      }),
      getCalendarDataProvider(),
      new AlertsSentry(Sentry.captureException.bind(Sentry)),
      logger,
      {
        environment: contentfulEnvId,
        space: contentfulSpaceId,
        accessToken: contentfulAccessToken,
      },
    )
  : calendarCreatedSquidexHandlerFactory(
      subscribeToEventChangesFactory(getJWTCredentialsAWS, logger, {
        asapApiUrl,
        googleApiToken,
        googleApiUrl,
        cms: 'squidex',
      }),
      unsubscribeFromEventChangesFactory(getJWTCredentialsAWS, logger, {
        googleApiUrl,
      }),
      getCalendarDataProvider(),
      new AlertsSentry(Sentry.captureException.bind(Sentry)),
      logger,
    );
```

`getCalendarDataProvider` is being used in some other places (`apps/crn-server/src/handlers/calendar/resubscribe-handler.ts` and `apps/crn-server/src/handlers/webhooks/webhook-events-updated.ts`) so this function still worth existing. So I think the ideia would be to call it directly inside the calendar handler factory, right?

looking at how the code would be with this change above, I don't think it would look much better, do you prefer to change to the snippet above?